src/org/labkey/test/util/FileBrowserHelper.java

```diff
@@ -880,56 +881,100 @@ public void openFolderTree()
         }
     }
 
-    private String stringOrNull(Object value)
+    private static String stringOrNull(Object value)
     {
         if (value == null)
             return null;
         return (String) value;
     }
 
-    public record FileDetailInfo(String fileName, String absoluteFilePath, String dataFileUrl, String webDavUrl, String webDavUrlRelative)
+    public record FileDetailInfo(String fileName, String absoluteFilePath, String dataFileUrl, String relativeFolder, String webDavUrl, String webDavUrlRelative)
     {
     }
 
-    public FileDetailInfo getFileDetailInfo(String containerPath, String fileName)
+    /**
+     * Queries the "exp"."files" table to gather metadata about a specific file.
+     *
+     * @param containerPath The container path in which to make the request.
+     * @param fileName The name of the file to find.
+     */
+    public static @Nullable FileDetailInfo getFileDetailInfo(String containerPath, String fileName)
+    {
+        return getFileDetailInfo(containerPath, fileName, null);
+    }
+
+    /**
+     * Queries the "exp"."files" table to gather metadata about a specific file. Optionally, a relativeFolder can be
+     * supplied to match against the "RelativeFolder" column. This is useful when looking for files in a specific
+     * subdirectory.
+     *
+     * @param containerPath  The container path in which to make the request.
+     * @param fileName       The name of the file to find.
+     * @param relativeFolder The expected value of the "RelativeFolder" column. If null, the column is not checked.
+     */
+    public static @Nullable FileDetailInfo getFileDetailInfo(String containerPath, String fileName, @Nullable String relativeFolder)
+    {
+        var fileInfos = getFileDetailInfos(containerPath, List.of(new Filter("Name", fileName)));
+
+        // "RelativeFolder" is not a filterable column on the query table
+        if (relativeFolder != null)
+            fileInfos = fileInfos.stream().filter(f -> relativeFolder.equals(f.relativeFolder)).toList();
```

Comment:
Maybe case insensitive check? Also is it worth checking with leading and trailing "/" ignored?